### PR TITLE
Feature/user notes

### PR DESCRIPTION
**Pull request by IdoElk**
Add user notes, backend only!

* Needs another frontend PR

### CODE REVIEW

File: lms/lmsweb/views.py [hunk 1]
```diff
@@ -452,6 +452,27 @@ def common_comments(exercise_id=None):
...
 
+@webapp.route('/user_note', methods=['POST'])
+@login_required
+@managers_only
+def user_note():
+    act = request.json.get('act')
+    note = request.json.get('note')
+    user = User.get_or_none(request.json.get('user_id', 0))
+
+    if not act or not user or note is None:
```
Comment (yammesicka) [inline lms/lmsweb/views.py:463]: Remove "not act" to get the right error message

File: lms/lmsweb/views.py [hunk 2]
```diff
@@ -452,6 +452,27 @@ def common_comments(exercise_id=None):
...
+        return fail(400, 'Invalid user_id or note')
+
+    if act == 'fetch':
+        return jsonify({user.note})
```
Comment (yammesicka) [inline lms/lmsweb/views.py:467]: Will it blend?
(Srsly though, you can just put `user.note` in there? Have you checked the API using Postman [or any similar tool]? What about the key?)

File: lms/lmsdb/models.py [hunk 3]
```diff
@@ -93,6 +93,7 @@ class User(UserMixin, BaseModel):
     mail_address = CharField(unique=True)
     password = CharField()
     role = ForeignKeyField(Role, backref='users')
+    note = CharField(max_length=4096, default='')
```
Comment (yammesicka) [inline lms/lmsdb/models.py:96]: It should be many to one - each user should many notes (Notes -> User)

Comment (IdoElk) [reply lms/lmsdb/models.py:96]: This was my initial thinking but then I was thinking about our actual usage.
Wouldn't we want to have a general note for a specific user? or am I missing the usecase?

**Pull request closed without merging**